src/demo/ui.jsx

```diff
@@ -543,8 +546,26 @@ class Words extends React.Component {
   constructor(props) {
     super(props);
 
-    // Randomize word choices and set in state.
-    const choices = _.shuffle(wordSet[getState().appMode].choices);
+    // Randomize word choices in each set, merge the sets, and set as state.
+    const appMode = getState().appMode;
+    const appModeWordSet = wordSet[appMode].choices;
+    let choices = [];
+    let maxSize = 0;
```

Comment:
Do we need to generalize this to the case where there are a different number of entries in each column? If we don't foresee needing it we could just instead validate that they're all equal to simplify.

Reply:
Fair enough. I tend to err on the side of handling array out of bounds (and like errors) even if we don't think they'll be necessary. I'll add a comment here.